packages/core/src/client.test.ts

```diff
@@ -1667,6 +1667,21 @@ describe('Client', () => {
     expect(fetch).toHaveBeenCalledTimes(2);
   });
 
+  test('Dispatch on bad connection', async () => {
+    const fetch = jest.fn(async () => {
+      throw { message: 'Failed to fetch' };
+    });
+    const mockDispatchEvent = jest.fn();
+    const client = new MedplumClient({ fetch });
+    client.dispatchEvent = mockDispatchEvent;
+    try {
+      await client.readResource('Patient', '123');
```

Comment:
To ensure that this test completely verifies the behavior you expect, you should put a `fail()` assertion on the next line after this one.  That way, if this line ever DOESN'T throw and enter the `catch` block, the test won't silently pass